### PR TITLE
Remove project's description field 

### LINKED ISSUE #3689: Remove project description field
I think we have good reason to just remove the description field. The dashboard page is public, but is not a reader focused page. This is extraneous UI for a maintainer dashboard page.

Having said that, we should evaluate what pulling this out looks like -- specifically:

* Can we simplify the import forms more by removing the description field?
* Does it make more sense to remove from modeling and just remove from forms and templates for now?
* What does the project dashboard page look like without a description block

### PR DESCRIPTION
**Pull request by dojutsu-user**
Closes #3689

### CODE REVIEW

File: requirements/pip.txt [hunk 1]
```diff
@@ -80,7 +80,6 @@ django-formtools==2.1
 # https://github.com/rtfd/readthedocs.org/issues/3999
 docker==3.1.3  # pyup: ignore
 
-django-textclassifier==1.0
```
Comment (dojutsu-user) [inline requirements/pip.txt]: This is not needed anymore.

File: readthedocs/templates/projects/index.rst.html [hunk 2]
```diff
@@ -1,11 +0,0 @@
-{% load projects_tags %}{{ project.name|heading|safe }}
```
Comment (dojutsu-user) [inline readthedocs/templates/projects/index.rst.html]: I didn't find any use case for this file.

**Comment by stsewd** [comment]
I think you need to keep this fields from the api response, see the discussion in the issue and the previous PR

**Comment by dojutsu-user** [comment]
@stsewd 
I didn't remove the field from the ProjectSerializer.
https://github.com/rtfd/readthedocs.org/blob/724c6cbddde1819d77521ce281b3d1097b85e5fe/readthedocs/restapi/serializers.py#L16-L32

File: readthedocs/restapi/serializers.py [hunk 3]
```diff
@@ -12,6 +12,15 @@
 
 class ProjectSerializer(serializers.ModelSerializer):
     canonical_url = serializers.ReadOnlyField(source='get_docs_url')
+    description = serializers.SerializerMethodField()
+
+    def get_description(self, obj):
+        """
+        Forcing the description field to be an empty string.
+
+        See https://github.com/rtfd/readthedocs.org/issues/3689.
```
Comment (dojutsu-user) [inline readthedocs/restapi/serializers.py]: Forcing the description to be an empty string in the API response. This was done as per [this comment](https://github.com/rtfd/readthedocs.org/issues/3689#issuecomment-371686362).

File: requirements/pip.txt (hunk 1 above)
Comment (davidfischer) [reply requirements/pip.txt]: I believe this is used for spam project detection. I do worry that we'd get more spam if we don't run any detection. Perhaps we could run it on the name of the project. @agjohnson?

Comment (dojutsu-user) [reply requirements/pip.txt]: @davidfischer 
There's an open PR for this. #5038

**Comment by humitos** [comment]
I remember that we talked in person about this, but I'm not really sure what was the path that we decided, @stsewd do you remember? Marking as Design Decision until then.

**Comment by stsewd** [comment]
I don't think we talked about this in person, but there was a conversation about this, what I remember (or I think I remember): hide the current field, the remove it from the db, but the api endpoints should remain (return empty or None)

File: readthedocs/templates/projects/index.rst.html (hunk 2 above)
Comment (dojutsu-user) [reply readthedocs/templates/projects/index.rst.html]: @stsewd is this file required?

**Pull request closed without merging**